Fix Docker build and runtime issues for cross-platform support

aashish24 opened the pull request:
Build fixes:
- Upgrade rasterio to 1.3.11 (fixes numpy==2.0.0rc1 build dependency)
- Add g++ and libgdal-dev for building rasterio from source on ARM
- Set GDAL_CONFIG env var for Fiona/tile2net builds

Runtime fixes:
- Add optional GPU profile for NVIDIA systems (celery-gpu service)
- Default celery runs CPU-only, use --profile gpu for GPU acceleration
- Fix web container to install npm dependencies inside container
- Use named volume for node_modules to avoid host/container arch mismatch
- Update Node.js to v22 (required by Vite 7.x)
- Remove platform: linux/amd64 from postgres for native ARM performance

Documentation:
- Update setup.md with GPU instructions and troubleshooting
- Improve formatting and add prerequisites section

annehaley commented on docs/setup.md:
```diff
@@ -1,36 +1,156 @@
+| Service | URL |
+|---------|-----|
+| Admin Panel | http://localhost:8000/admin/ |
+| User Interface | http://localhost:8080/ |
```
I like this little table. We could also add `http://localhost:8000/api/docs/swagger/`, the API reference page.

augustposch replied:
I got as far as seeing the Admin Panel login page and seeing the User Interface login page. As a next step, can someone assign me a login please?

annehaley commented on docs/setup.md:
````diff
@@ -1,36 +1,156 @@
+
+```bash
+# Apply database migrations
+docker compose run --rm django ./manage.py migrate
````
While working with August to get his local instance running, we learned that on Windows, invoking `./manage.py` directly encounters a carriage return issue. Invoking it with `python manage.py` instead works fine. Let's replace all of the `./manage.py` invocations in these instructions with `python manage.py`.

zachmullen commented:
I just did a fresh `docker compose build --no-cache` on my Apple Silicon machine and it worked without needing to set any platforms to `amd64`. Awesome!

annehaley commented on docker-compose.override.yml:
```diff
@@ -44,26 +45,61 @@ services:
     volumes:
       - .:/opt/geoinsight-server
-    # NVIDIA driver and increased memory limit are required for tile2net inference
+    shm_size: 1gb
```
After #248, we no longer require a GPU-enabled container for the `tie2net` task. We can still keep the option around for other accelerated inferencing, though.

annehaley commented on docs/setup.md:
````diff
@@ -1,36 +1,163 @@
+# Install frontend dependencies
+cd web && npm i
+cd ..
+```
````
The installation of node modules is now included in the web container's command, so we can omit this section from the setup guide.

aashish24 commented:
@annehaley thanks for your review. I addressed your comments. I kept reference to node installation in case developers might work on frontend outside containers

augustposch reviewed (changes requested):
<img width="1440" height="257" alt="image" src="https://github.com/user-attachments/assets/4d6a479b-66c2-4697-af81-471b6c567f3f" />
This time `docker compose up` didn't work for me due to missing semicolon. Skimming the yaml files, I'm not sure why there would be a problem with a semicolon in line 48. Any suggestion for me?
